joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/DynamicTypeHintFullNamePass.scala

```diff
@@ -47,12 +48,32 @@ class DynamicTypeHintFullNamePass(cpg: Cpg) extends CpgPass(cpg) {
     } {
       importDetails match {
         case (Some(importedEntity), Some(importedAs)) =>
-          val typeFullName = typeHint.replaceFirst(Pattern.quote(importedAs), importedEntity)
-          diffGraph.setNodeProperty(param, PropertyNames.DYNAMIC_TYPE_HINT_FULL_NAME, typeFullName)
+          setTypeHints(diffGraph, param, typeHint, importedAs, importedEntity)
         case _ =>
       }
 
     }
 
   }
+
+  private def setTypeHints(
+    diffGraph: BatchedUpdate.DiffGraphBuilder,
+    node: StoredNode,
+    typeHint: String,
+    alias: String,
+    importedEntity: String
+  ) = {
+    val typeFullName = typeHint.replaceFirst(Pattern.quote(alias), importedEntity)
+    val typeFilePath = typeFullName.replaceAll("\\.", File.separator)
+    val pythonicTypeFullName = typeFullName.split("\\.").lastOption match {
+      case Some(typeName) =>
+        typeFilePath.stripSuffix(s"${File.separator}$typeName").concat(s".py:<module>.$typeName")
+      case None => typeFullName
+    }
+    cpg.typeDecl.fullName(s".*$pythonicTypeFullName").l match {
```

Comment:
`fullName` or `fullNameExact`?

Reply:
Definitely full name, since `app.foo.bar` is imported as `foo.bar` so this catches that, but I think Windows is failing since I'm injecting that variable into the regex raw.